### ISSUE TITLE
unknown service lnrpc WalletUnlocker

### ISSUE THREAD

**Issue by torkelrogstad** [lightninglabs/lightning-app#513]
### Description 
Unlocking the wallet after it's been locked (excluding the initial unlock at startup) fails with error `unknown service lnrpc.WalletUnlocker`. 

### Steps to reproduce the behavior
Open the app, and unlock the wallet. Make a change in any file, forcing the app to recompile. Upon seeing the "UNLOCK WALLET" screen, enter any password (correct or incorrect). 

### Expected behavior
The wallet should unlock.

### Actual behavior
If a log call is inserted in the catch block of `unlockWallet` in `src/action/wallet.js`, the error `{"code":12,"details":"unknown service lnrpc.WalletUnlocker","metadata":{"_internal_repr":{}}}` is printed.

**Comment by tanx** [comment]
> Open the app, and unlock the wallet. Make a change in any file, forcing the app to recompile. Upon seeing the "UNLOCK WALLET" screen, enter any password (correct or incorrect).

Yeah. Hot reloading an unlocked wallet doesn't work yet during development.

**Comment by torkelrogstad** [comment]
Any idea why? Is it a problem with gRPC, or something fixable here?

**Comment by Roasbeef** [comment]
> {"code":12,"details":"unknown service lnrpc.WalletUnlocker","metadata":{"_internal_repr":{}}}

If the app _frontend_ has been restarted, but `lnd` itself hasn't, then this would make sense. We essentially use two distinct RPC services to accept auth/pass info vs accepting regular RPC calls. Once the wallet has been unlocked or created, we tear down the initial RPC server in favor of the fully featured one.

**Comment by tanx** [comment]
I think the main issue is on the app side, since electron hot reloads the updated JS bundle and we might not handle that case correctly. Hasn’t been a priority, since we do the UI development in React storybook and JS logic development with unit/integration tests. So you shouldn’t need to iterate/develop with the app open.

**Comment by Roasbeef** [comment]
It's a gRPC error in the current form as is, since we tear down the auth server (can't unlock the wallet after it's already been unlocked). It may be possible for hot reloading also to re-launch the binary? I think with the promoted dev style (story book, etc) as you said, this isn't necessarily an issue.

**Comment by coder82** [comment]
Any reason why you don't answer what happens right in the response? If wallet is already unlocked I get this response from the lncli:

> lncli unlock
> [lncli] Wallet is already unlocked

I would expect the same happening on the gRPC side, why 2 different behaviors?
In Java I get an exception which I need to catch every time I restart my app.

> io.grpc.StatusRuntimeException: UNIMPLEMENTED: unknown service lnrpc.WalletUnlocker

Why don't you rather populate the response which is now empty?
`message UnlockWalletResponse {/*tell us something*/}`

Unless you have a good architectural reason for tearing down the wallet rpc and not keeping it alive, especially after I realized you keep the wallet service separated from the rest of the api.
